content/en/docs/ops/ambient/upgrade/helm-upgrade/index.md

```diff
@@ -72,6 +72,12 @@ $ helm upgrade istiod istio/istiod -n istio-system
 
 The ztunnel DaemonSet is the L4 node-proxy component of ambient.
 
+{{< warning >}}
+As ambient is not yet stable, the following statement is not a compatibility guarantee and is subject to change, or removal. Prior to reaching stable status, this component and/or the control plane may receive breaking changes that prevent compatibility between minor versions.
+{{< /warning >}}
+
+The ztunnel at version 1.x is generally compatible with control plane at version 1.x-1, 1.x, and 1.x+1, which means the ztunnel DaemonSet and the Istio control plane can be upgraded independently and in any order, as long as their version difference is within one minor version.
```

Comment:
I'm still concerned about `can be upgraded independently and in any order` as this is not something we test in the past.  I'd prefer to only have the following, unless we can test 1.x-1 and have automated tests:

The ztunnel at version 1.x is generally compatible with control plane at version 1.x, and 1.x+1

Reply:
It is something that is generally followed with other components, IE CNI. I think until after it's in beta, it is sufficient as is and can be discussed maybe during the Ambient meeting what compatibility requirements we need on ztunnel/CNI.

Reply:
I 'd think istio-cni is different comparing with ztunnel as ztunnel is data plane.  we always recommend users to update CP first then DP, I'd rather avoid sending confusion messages. 

I'm tagging more folks to comment @istio/wg-environments-maintainers.

Reply:
The goal with WDS/ztunnel and ambient in general is to make the dataplane and CP independent. Part of that independence means they do not have an upgrade ordering dependency.

If our tests assume a specific order, they probably shouldn't - but I don't think we have any ambient upgrade tests, so it's likely moot.

When we do have them, they should test either order, for robustness.

Reply:
The challenge is we do change WDS API sometimes, may be not a whole lot, but if we do change it from n to n+1, how would we ensure that n still work with n+1's ztunnel?

Reply:
+1, prefer not to have n+1 support, at least until we are more stabilized

Reply:
yeah and if users request for it.

Reply:
KK, will update to N, N-1 for now.

I do think we should definitely start to think about having tests that do both permutations tho - even if users don't ask for it, it will catch bugs, incompatibilities, and assumptions in CI that we are likely to otherwise miss.

Reply:
Also just to clarify - if we want to enforce CP first then zt/CNI, it's `N+1` on the control plane side actually:

> The ztunnel at version 1.x is generally compatible with _control plane at version_ 1.x+1 and 1.x, which means the control plane must be upgraded first.

Reply:
That is correct. Your update LG

Reply:
It is not only about testing, it is also about ensure ztunnel code works with older CP, which is additional work and baggage to carry when there is WDS change.  We don't support updating DP first with sidecars, and I don't recall anyone complains.  People usually complains who they follow the doc and stuff doesn't work.